pretyping/typeclasses.ml

```diff
@@ -442,19 +442,20 @@ let add_class cl =
 
 let instance_constructor (cl,u) args =
   let lenpars = List.count is_local_assum (snd cl.cl_context) in
+  let open EConstr in
   let pars = fst (List.chop lenpars args) in
     match cl.cl_impl with
       | IndRef ind -> 
         let ind = ind, u in
-          (Some (applistc (mkConstructUi (ind, 1)) args),
-	   applistc (mkIndU ind) pars)
```

Comment:
Should we maybe have econstr applistc instead?

Reply:
I would be more in favor of removing one of the two, having two versions seems a bit pointless, right? Are they used in some particular combinator?